scripts/detect-allocators.sh

```diff
@@ -0,0 +1,87 @@
+#!/bin/bash
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the Apache 2.0 License.
+
+# Make sure that the resulting binary
```

Comment:
I think we could simplify and have check-link-snmalloc and return non-zero if not. Also we can objdump -C rather than deal with mangled symbols.